Geometry_Engine/Query/Distance.cs

```diff
@@ -142,6 +142,215 @@ public static double Distance(this Point point, NurbsCurve curve)
         }
 
 
+        /***************************************************/
+        /****       Public Methods - Curve/Curve        ****/
+        /***************************************************/
+
+        public static double Distance(this Line curve1, Line curve2, double tolerance = Tolerance.Distance)
```

Comment:
Question regarding all of those methods. Should we not be able to get away with just one, calling the interface methods for the Curve proximity, as :

```
 public static double Distance(this ICurve curve1, ICurve curve2, double tolerance=Tolerance.Distance)
        {
            BH.oM.Reflection.Output<Point, Point> results = curve1.ICurveProximity(curve2,tolerance);
            return results.Item1.Distance(results.Item2);
        }
```

Reply:
Was thinking about this, because all of them is basically the same, but wasn't sure if this shouldn't be split like the rest. I can of course change this to the form accepting (ICurve, ICurve) if this is preferred way. Will have to just exclude NurbsCurve and Elipse.

Reply:
I believe @IsakNaslundBh is right - probably even NurbsCurve and Ellipse could be called dynamically -> it will hit against the NotImplemented `CurveProximity` method.

Reply:
Can understand why you added them in, but the reasons they are split in general is due to necessity rather than that we want to. Always better to keep as few as we can (while still keeping all functionality) to keep both the code more maintainable as well as avoiding to flood the UI. I would go for updating to this.

Regarding the ellipse and NurbsCurve, good to get them into the curve proximity methods and just returning NotImplemented exception as well as `[NotImplmented]` attribute (this might already have been done). Then this will return a correct error, and will start working straigh away when/if they get added.

Reply:
Okay, will correct in a second then.

Reply:
Done.